torch/_dynamo/output_graph.py

```diff
@@ -814,18 +865,53 @@ def remove_unused_graphargs(self) -> None:
                 elif node.op == "call_function" and node.target is operator.getitem:
                     self.remove_node(node)
 
-        for i, node in enumerate(self.placeholders):
-            if not node.users:
-                log.debug(
-                    "REMOVE UNUSED GRAPHARG %s", node.meta["grapharg"].source.name()
-                )
-                # I'm not really sure why you need to delete these from the
-                # node since the node is going to get removed
-                if "example_value" in node.meta:
-                    del node.meta["example_value"]
-                del node.meta["grapharg"]
-                self.remove_node(node)
-                self.real_value_cache.pop(node, None)
+        def placeholder_binds_symbol(node):
+            arg = node.meta["grapharg"]
+            example = arg.example
+            if isinstance(example, torch.SymInt) and isinstance(
+                example.node.expr, sympy.Symbol
+            ):
+                return example.node.expr
+            return None
+
+        def remove_unused(node):
+            log.debug("REMOVE UNUSED GRAPHARG %s", node.meta["grapharg"].source.name())
+            # I'm not really sure why you need to delete these from the
+            # node since the node is going to get removed
+            if "example_value" in node.meta:
+                del node.meta["example_value"]
+            del node.meta["grapharg"]
+            self.remove_node(node)
+            self.real_value_cache.pop(node, None)
+
+        used_symbols = set()
+        recheck_placeholders = []
+        for node in self.placeholders:
+            binds_symbol = placeholder_binds_symbol(node) is not None
+            # Don't delete symbol bindings yet
+            if binds_symbol:
+                if not node.users:
+                    recheck_placeholders.append(node)
```

Comment:
When is this the case? why do we need to recheck? is this more like, unused_bound_symbols?

Reply:
They're all the placeholders that we need to check again after we've found out what symbols we need to bind. They are not unused bound symbols because they might be implicitly "used" by other tensors which refer to their symbol.